src/main/java/org/jalasoft/moi/controller/services/FileService.java

```diff
@@ -119,12 +120,59 @@ public Parameters saveFileB64(String name, String codeB64, Long projectId) throw
      * @param code saves the code inside the file
      * @param projectId gets the project path by project id
      */
-    private void saveFile(String name, String code, Long projectId) throws IOException {
+    private String saveFile(String name, String code, Long projectId) throws IOException {
         Project projInfo = projectRepository.findById(projectId).get();
         String filePath = projInfo.getPath()+"/"+ name + projInfo.getLanguage().getFileExtention();
         FileWriter codeWriter = new FileWriter(filePath);
         codeWriter.write(code);
         codeWriter.close();
+        return filePath;
+    }
+
+    /**
+     * Updates a new file with name and code to set the file properties in base 64.
+     *
+     * @param fileId gets the file information by its id
+     * @param name updates the file name in the project
+     * @param codeB64 updates the code inside the file in base 64
+     * @return a parameters object
+     */
+    public void updateFileB64(Long fileId, String name, String codeB64) throws IOException {
+        byte[] byteArray = Base64.decodeBase64(codeB64.getBytes());
+        String code = new String(byteArray);
+        updateFile(fileId, name, code);
+    }
+
+    /**
+     * Updates a new file with name and code to set the file properties in a new params object.
+     *
+     * @param fileId gets the file information by its id
+     * @param name updates the file name in the project path
+     * @param code updates the code inside the file
+     */
+    private String updateFile(Long fileId, String name, String code) throws IOException {
+        deleteFile(fileId);
+        FileCode fileCode = fileRepository.findById(fileId).get();
+        Project projInfo = fileCode.getProject();
+        String filePath = projInfo.getPath()+"/"+ name + projInfo.getLanguage().getFileExtention();
+        FileWriter codeWriter = new FileWriter(filePath);
+        codeWriter.write(code);
+        codeWriter.close();
+        return filePath;
+    }
+
+    /**
+     * Deletes a file from the project foler.
+     *
+     * @param fileId gets a file information by its id
+     */
+    private void deleteFile(Long fileId) throws IOException {
+        FileCode fileCode = fileRepository.findById(fileId).get();
+        Project projInfo = fileCode.getProject();
```

Comment:
It's a good practice look if exist the entity in the database:
you can use `projectRepository.findById(projectId).isPresent()` to look for that, since the demo is coming soon we can update this to next version.